### PR TITLE
Add common R action

### PR DESCRIPTION
**Pull request by AlexAxthelm**
Add standard R checking workflows from the RMI-PACTA/actions repo.

Relates to:
- [ ] https://github.com/RMI-PACTA/r2dii.data/issues/386
- [ ] https://github.com/RMI-PACTA/r2dii.analysis/issues/501
- [ ] https://github.com/RMI-PACTA/r2dii.plot/issues/567

### CODE REVIEW

**Comment by cjyetman** [comment]
probably doesn't make sense to add this without removing the existing R CMD checks

File: .github/workflows/R.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+name: R
+
+jobs:
+  R-package:
+    name: R Package Checks
+    uses: RMI-PACTA/actions/.github/workflows/R.yml@main
+
+  dev-r-cmd-check:
+    name: R CMD Check (dev versions)
+    uses: RMI-PACTA/actions/.github/workflows/R-CMD-check.yml@add-extra-repos-to-rcmdcheck
```
Comment (jdhoffa) [inline .github/workflows/R.yml:27]: This action still depends on a PR branch, please merge that branch first

**Review by jdhoffa — CHANGES_REQUESTED** [review]

**Comment by jdhoffa** [comment]
Depends on https://github.com/RMI-PACTA/actions/pull/90

File: R/prioritize.R [hunk 2]
```diff
@@ -21,7 +21,29 @@
...
+#'     * If you are happy with the match, set the `score` value to `1`.
+#'     * Otherwise set or leave the `score` value to anything other than `1`.
+#' * Save the edited file as, say, _valid_matches.csv_.
+#' 
+#' Re-read the edited file (validated) with:
+#' 
+#' ```
+#' # Reading from current working directory
+#' valid_matches <- readr::read_csv("valid_matches.csv")
+#' ```
```
Comment (jdhoffa) [inline R/prioritize.R:46]: Please move this change (and the other content/ not action related file commits) to a separate PR (to keep this PR focused on changing only the workflow files)

**Review by jdhoffa — CHANGES_REQUESTED** [review]

File: R/prioritize.R (hunk 2 above)
Comment (AlexAxthelm) [reply R/prioritize.R]: https://github.com/RMI-PACTA/r2dii.match/pull/483

**Comment by AlexAxthelm** [comment]
Note that with this PR, we're testing the HEAD (latest version of `main` or PR) against both the CRAN and latest `main` versions of all dependencies.

Note that when we are testing against the dev versions of the dependencies, it is for the RMI-PACTA packages only (for example, `dplyr` is still the CRAN version).

**Comment by AlexAxthelm** [comment]
@jdhoffa I believe the requested changes were made in #483

**Review by jacobvjk — APPROVED** [review]

**Review by jdhoffa — APPROVED** [review]
lgtm!

**Pull request merged**